packages/core/src/queries/user.ts

```diff
@@ -90,6 +90,30 @@ export const findTotalNumberOfUsers = async () => getTotalRowCount(table);
 
 const findUserMany = buildFindMany<CreateUser, User>(pool, Users);
 
+export const searchUsers = async (limit: number, offset: number, search: string) =>
+  pool.many<User>(
+    sql`
+      select ${sql.join(Object.values(fields), sql`,`)}
+      from ${table}
+      where ${fields.primaryEmail} like ${'%' + search + '%'}
+      or ${fields.primaryPhone} like ${'%' + search + '%'}
+      or ${fields.username} like ${'%' + search + '%'}
+      or ${fields.name} like ${'%' + search + '%'}
+      limit ${limit}
+      offset ${offset}
+    `
+  );
+
+export const findTotalNumberOfUsersWithSearch = async (search: string) =>
+  pool.one<{ count: number }>(sql`
+    select count (*)
+    from ${table}
+    where ${fields.primaryEmail} like ${'%' + search + '%'}
+    or ${fields.primaryPhone} like ${'%' + search + '%'}
+    or ${fields.username} like ${'%' + search + '%'}
+    or ${fields.name} like ${'%' + search + '%'}
```

Comment:
this part can be reused

Reply:
also you can leverage `packages/core/src/database/find-many.ts:33-36` for conditional where clause, and add the feature (like clause) to `buildFindMany()` later (eng quality phase 4).

Reply:
then you don't need two separate functions and `?:` in the route.

Reply:
please add tests after we settle on implementation

Reply:
updated.
and the search feature will be added to `buildFindMany` in LOG-1599